calus.js

```diff
@@ -1,228 +1,264 @@
-            // date objects)
-            availableDates: options.availableDates || [],
-            // currently selected date. this is reset when changing available dates
-            selected: null,
-            // whether to show all the months in a column, or a single month with
-            // controls to change which month is shown
-            displayInColumn: options.displayInColumn || false,
-
-            // linear view for when showing a free flowing calendar
-            linearDates: options.linearDates || false,
-
-            // which month is currently shown on screen (only used when
-            // `displayInColumn` is false)
-            currentDisplayedMonth: DateTime.local().startOf('month'),
-            // force the day to start on sunday instead of default monday
-            weekStartsOnSunday: options.weekStartsOnSunday || false,
-
-            // callback for when an available date is clicked
-            onSelect: options.onSelect || function (day) { },
-            // callback for when selected month is changed with button
-            onChangeMonth: options.onChangeMonth || function(prev, current) { },
-        },
-        computed: {
-            now: () => DateTime.local(),
-            firstAvailable: function () {
-                return this.availableDates.length ? this.availableDates[0] : this.now
+import { DateTime, Settings } from 'luxon'
+
+export default {
+    props: {
+        timeZone: String,
+        availableDates: {
+            type: Array,
+            default: function () {
+                return [];
             },
-            lastAvailable: function () {
-                return this.availableDates.length
-                    ? this.availableDates[this.availableDates.length - 1]
-                    : this.now.plus({ months: 2 })
+        },
+        displayInColumn: {
+            type: Boolean,
+            default: false,
+        },
+        linearDates: {
+            type: Boolean,
+            default: false,
+        },
+        currentDisplayedMonth: {
+            type: Object,
+            default: function () {
+                return DateTime.local().startOf("month");
+            },
+        },
+        weekStartsOnSunday: {
+            type: Boolean,
+            default: false,
+        },
+        onSelect: {
+            type: Function,
+            required: true,
+        },
+        onChangeMonth: {
+            type: Function,
+        },
+        allowScrollingOutsideOfDateRange: {
+            type: Boolean,
+            default: true,
+        },
+        monthControlsClasses: {
+            type: Array,
+            default: function () {
+                return [];
             },
-            months: function () {
-                let months = []
-                let date = null
+        },
+    },
+    data: function () {
+        return {
+            datesAvailable: [],
+        };
+    },
```

Comment:
You don't have any reference to `availableDates` the property being passed in
You should just remove this one and rename `availableDates` to `datesAvailable` in as the property

Reply:
removed